tests/AsyncEventDispatcherTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ArchiPro\EventDispatcher\Tests;
+
+use ArchiPro\EventDispatcher\AsyncEventDispatcher;
+use ArchiPro\EventDispatcher\ListenerProvider;
+use ArchiPro\EventDispatcher\Tests\Fixture\TestEvent;
+use PHPUnit\Framework\TestCase;
+use Revolt\EventLoop;
+
+/**
+ * Test cases for AsyncEventDispatcher.
+ *
+ * Verifies the asynchronous event dispatching functionality, including:
+ * - Multiple listener execution
+ * - Event propagation stopping
+ * - Handling events with no listeners
+ *
+ * @covers \ArchiPro\EventDispatcher\AsyncEventDispatcher
+ */
+class AsyncEventDispatcherTest extends TestCase
+{
+    private ListenerProvider $listenerProvider;
+    private AsyncEventDispatcher $dispatcher;
+
+    /**
+     * Sets up the test environment before each test.
+     */
+    protected function setUp(): void
+    {
+        $this->listenerProvider = new ListenerProvider();
+        $this->dispatcher = new AsyncEventDispatcher($this->listenerProvider);
+
+        EventLoop::setErrorHandler(function (\Throwable $err) {
+            throw $err;
+        });
+
+    }
+
+    /**
+     * Tests that multiple listeners for an event are executed.
+     */
+    public function testDispatchEventToMultipleListeners(): void
+    {
+        $results = [];
+        $completed = false;
+
+        $this->listenerProvider->addListener(TestEvent::class, function (TestEvent $event) use (&$results) {
+            EventLoop::delay(0.1, fn () => null);
```

Comment:
```
delay(0.1);
echo 'after waiting 0.1 seconds';
```

Reply:
I don't really want my test to be echoing things to the console  though.

Reply:
It was just pseudocode, perhaps I misunderstood your test. I was just pointing out that the EventLoop::delay() here does almost nothing.. I thought you were specifically trying to block execution with a sleep (delay in the context of amphp).